### PR TITLE
Fix bad message break notebook

### PR DESCRIPTION
**Pull request by Carreau**
Would need a test, and probably a way to display the message to the user
if it ever happen.

---

Who did the WrappedError ? was it @jasongrout ? @jdfreder ?
Do any of you rely on the `.catch()` behavior ?

How would you display the error if any ? Dialog ? 
Or  red badge in notification area which on click show a dialog with the full error message ?

### CODE REVIEW

File: notebook/static/services/kernels/kernel.js [hunk 1]
```diff
@@ -882,8 +882,14 @@ define([
...
         this._msg_queue = this._msg_queue.then(function() {
             return serialize.deserialize(e.data);
-        }).then(function(msg) {return that._finish_ws_message(msg);})
-        .catch(utils.reject("Couldn't process kernel message", true));
+        }).then(function(msg) {
+            try {
+                return that._finish_ws_message(msg);
+            } catch (e) {
+                console.error("Couldn't process kernel message", e)
+                return Promise.resolve()
```
Comment (minrk) [inline notebook/static/services/kernels/kernel.js]: Don't forget your semicolons...

Comment (Carreau) [inline notebook/static/services/kernels/kernel.js]: I prefer [greek question mark]9http://www.fileformat.info/info/unicode/char/037e/index.htm)

**Comment by jdfreder** [comment]
I wrote the WrappedError implementation.  I never display it to the user.  It's only ever logged to the console.  It maintains a nice stack, which is lost in complex promise operations.  In the widgets we use `catch` and `then` extensively.

If you need to display an error, I'd continue logging the error to the console with the stack but additionally display a human digestible error, depending on the context.

File: notebook/static/services/kernels/kernel.js [hunk 2]
```diff
@@ -882,8 +882,14 @@ define([
...
         this._msg_queue = this._msg_queue.then(function() {
             return serialize.deserialize(e.data);
-        }).then(function(msg) {return that._finish_ws_message(msg);})
-        .catch(utils.reject("Couldn't process kernel message", true));
+        }).then(function(msg) {
+            try {
+                return that._finish_ws_message(msg);
+            } catch (e) {
+                console.error("Couldn't process kernel message", e);
+                return Promise.resolve();
```
Comment (jdfreder) [inline notebook/static/services/kernels/kernel.js]: So the difference here is that you want to promise chain to continue, even if there is a failure?

If so, you can continue to use catch, just don't use the utils.reject function.

Comment (jdfreder) [inline notebook/static/services/kernels/kernel.js]: Try this
```js
        }).then(function(msg) {return that._finish_ws_message(msg);})
        .catch(function() {console.error("Couldn't process kernel message") });
```

**Comment by Carreau** [comment]
Done.

**Comment by jdfreder** [comment]
This looks good to me, I'm waiting to see what Mr. Travis thinks...

**Comment by jdfreder** [comment]
Thanks @Carreau !

**Comment by Carreau** [comment]
Thanks for review on tips !

**Pull request merged**